docs/kubernetes.md

```diff
@@ -26,13 +26,11 @@ Custom Resource Definitions must have a **group** and **version**, which are use
 ### CRDs & grafana-app-sdk
 
 When you author kinds and use `grafana-app-sdk generate` to [generate code](code-generation.md), you'll get a CRD file for each kind as well. 
-The CRD file will contain the entire lineage expressed in openAPI format, and can be used to create your kind as a CRD in a kubernetes cluster. 
+The CRD file will contain the entire lineage expressed in openAPI format, and can be used to create your kind as a CRD in a kubernetes cluster.
 The generated `resource.Schema` is then used to identify the **group**, **version**, and **kind** of your CRD when interacting with kubernetes. 
 You can then use a `k8s.ClientRegistry` to generate clients which can translate the CRDs in the cluster into the generated go type. 
 
-The reason the client intermediary is necessary is twofold:
-1. It allows thema to be inserted at the unmarshal point
-2. The format of metadata in your authored kind doesn't exactly match kubernetes' metadata format--the kind has more metadata, which is encoded and decoded in kubernetes' annotations for the object. The client performs this translation.
+The client intermediary is necessary because the metadata format in your authored kind doesn't exactly match Kubernetes' metadata format. The kind has more metadata, which is encoded and decoded in the Kubernetes object's annotations, and the client performs this translation.
```

Comment:
This is somewhat out-of-date, as the client no longer encodes/decodes the annotations metadata, the generated object implementation has getters/setters for the metadata fields that encode it to the annotations. The generated object now should match the kubernetes expectation, but we use the client for two main reasons:
1. It allows us to introduce efficiencies under the hood transparently to the app author
2. It ensures that we use the encoding/decoding process in the kind's `resource.Codec`, rather than a straight JSON marshal/unmarshal

At this point, the client intermediary isn't strictly necessary for generated `resource.Object` implementations (though it may still be necessary for hand-written ones that use custom codecs), but is still favored because of the above two reasons.